dvc/repo/experiments/executor/base.py

```diff
@@ -657,19 +659,22 @@ def _repro_dvc(  # noqa: C901
             else:
                 os.chdir(dvc.root_dir)
 
+            args_path = os.path.join(dvc.tmp_dir, cls.PACKED_ARGS_FILE)
+            if os.path.exists(args_path):
+                _, kwargs = cls.unpack_repro_args(args_path)
+            # set env vars using saved config; needed for dvclive to read config
+            for k, v in kwargs.get("run_env", {}).items():
+                os.environ[k] = v
+            dvc_studio_config = get_studio_config(dvc.config.get("studio"))
```

Comment:
We are unnecessarily calling `get_studio_config`.

We do need to extract some info from the `run_env` (mainly the `studio_repo_url` because in the queue it can't be inferred) but I think it would be better to not modify `os.environ` here with a bunch of unrelated/unneeded things.

It would be better to pass as arguments, like:

```suggestion
            run_env = kwargs.get("run_env", {})
            dvc_studio_config = {
                "offline":  run_env.get(DVC_STUDIO_OFFLINE, False),
                ...
                ...
            }
```

Reply:
Okay, I think I misunderstood. I can drop changing the env. 

Still need to think through a couple other issues:
1. How we should prioritize info set at the time of queueing vs at the time of starting the experiment. Setting the `repo_url` is needed, but it's not clear how to handle other info.
2. How to ensure the configuration matches between DVC start/done calls and DVCLive data calls.

Reply:
> Setting the repo_url is needed, but it's not clear how to handle other info.

Which other info is not clear? The offline toogle?
I don't see much of an issue but we could just choose for the special case of offline to always respect the env var first

> How to ensure the configuration matches between DVC start/done calls and DVCLive data calls.

If you pass the contents of `run_env` to `post_live_metrics` as the `dvc_studio_config`, they will take precedence and match what DVCLive will use (because those same contents will be set as env vars int he stage)